src/txn/txn.c

```diff
@@ -1574,7 +1574,8 @@ __wt_txn_prepare(WT_SESSION_IMPL *session, const char *cfg[])
              * simply discarded when we find them.
              */
             for (tmp = upd->next; tmp != NULL && tmp->txnid == upd->txnid; tmp = tmp->next)
-                if (tmp->type != WT_UPDATE_RESERVE) {
+                if (tmp->type != WT_UPDATE_RESERVE &&
+                  !F_ISSET(tmp, WT_UPDATE_RESTORED_FAST_TRUNCATE)) {
```

Comment:
I'd suggest a comment here. Something like:
```
1574c1574,1576 
<            * simply discarded when we find them.
---            
>            * simply discarded when we find them. Also ignore updates created by instantiating
>            * fast truncation pages, they aren't linked into the transaction's modify list and so
>            * can't be considered.
```

Reply:
@raviprakashgiri29 the comment given by Keith to update the comment with his suggested text is not yet updated in the latest code. Maybe you missed to handle it?

Reply:
@kommiharibabu Fixed comments.